src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -357,7 +357,11 @@ function VisualizeAppController(
     };
 
     $scope.showQueryBarTimePicker = () => {
-      return vis.type.options.showTimePicker;
+      // tsvb loads without an indexPattern initially (TODO investigate).
+      // hide timefilter only if timeFieldName is explicitly undefined.
+      let hasTimeField = true;
+      if ($scope.indexPattern) hasTimeField = !!$scope.indexPattern.timeFieldName;
```

Comment:
can we use the `{}` + newline syntax for readability? Also, this could all be a ternary statement, right?
`const hasTimeField = $scope.indexPattern ? !!$scope.indexPattern.timeFieldName : true`